lib/livebook/runtime/evaluator.ex

```diff
@@ -1081,6 +1100,27 @@ defmodule Livebook.Runtime.Evaluator do
 
   defp pythonx_version(), do: List.to_string(Application.spec(:pythonx)[:vsn])
 
+  defp after_evaluation(:python) do
+    if ensure_pythonx() == :ok do
+      # With matplotlib the charts are built imperatively, by modifying
+      # a global figure state. We clear the global state after the
+      # evaluation, otherwise re-evaluating cells draws on top of the
+      # previous figure. We do this only if matplotlib is imported.
```

Comment:
Couldn't we do this after we call Kino.Render and we detect a matplotlib thingy?

Reply:
We could, but it can lead to weird cases, because the cleanup would happen only once the plot is returned, which is less precise. For example imagine we have two cells:

```python
import matplotlib.pyplot as plt

plt.plot([1, 2], [1, 2])
```

```python
plt.gcf()
```

If we evaluate the first one twice it alters the same global state, and it is cleared only once the second cell is evaluated rendering the plot. In this specific case the line color would depend on how many times you evaluated the first cell 😄

So encapsulating the statefulness to a single evaluation sounds like the most sane approach. WDYT?

(Yes, I do hate that we need to special case matplotlib in Livebook :D)